### DISCUSSION TITLE
Add Hard Disk Noises

### DISCUSSION THREAD

**Discussion by navjack** [86Box/86Box#2919] [Ideas]
I would love there to be something like in Winuae where the disks make noise.

**Comment by Banjo-Oz** [comment]
I've been begging PCem to add this. Would *love* it in 86Box, even if just floppy noises!

**Comment by pioneerman2k1** [comment]
HDD and FDD noises would be so great!

**Comment by danrevella** [comment]
At least for floppy, please :-)

**Comment by dgurney** [comment]
There are several difficulties in a request like this:
1) There would need to be several different hard disk sounds for different era disks. After all, MFM drive sounds simply do not fit a Pentium machine.
2) They would need to be tied perfectly to disk activity to sound non-artificial.

**Comment by danrevella** [comment]
@dgurney
you're right, BTW, IMO (at least for floppy) artificial is always better then no sound.....

**Comment by ghonore64** [comment]
I just found this video:

https://www.youtube.com/watch?v=LP2gTlpeAN8

Sounds awesome, doesn't it?

**Comment by danrevella** [comment]
It's amazing, fabolus!!!!

**Comment by ghost** [comment]
> 
> 
> @dgurney
> you're right, BTW, IMO (at least for floppy) artificial is always better then no sound.....

Nope, I think it isn't. Anyway, developers must add controls to enable HDD sounds, FDD sounds, ZIP sounds, and etc.

**Comment by ghost** [comment]
Not a bad idea but that's something for the future. It'll require for someone to friendly capture, edit and provide many samples of FDD, HDD activity noises but also as dgurney referenced they would need to be perfectly tied to the activity else they'll sound weird.

**Comment by dhrdlicka** [comment]
ref #1273

**Comment by ghost** [comment]
I think we should make an audio synthesizer of drives we want, otherwise this feature will be less cool — once WAV files are used. Also, they'll increase the size of the emulator distribution kit. Buffer size must be specified when calling the audio synthesis function. WAV files usage is purely demonstrated in following video: https://www.youtube.com/watch?v=LP2gTlpeAN8

In my opinion, prototype of the functions will be like this:
```
SoundSynthesis. [ INSERT_DRIVE_NAME_HERE ] (int model, int micro_action, int buf_size, short int volume);
```

**Comment by ghost** [comment]
@tiseno100, If the usage of WAV files is predefined, could I help with the implementation? I have advanced skills in audio mastering (editing).

**Comment by edescourtis** [comment]
https://youtu.be/eSNqzTwHiuU

**Comment by Fenix770** [comment]
It would require to emulate the right read/write speeds of these different hard disks, because the sound must represent the disk activity correctly. I think the proper speed emulation is more important than sound, because when I defrag a disk with NU on XT machines with MFM controllers, the hard disk speed is very close to a RAM drive.
I believe It would be possible only for XTA-IDE and very early IDE hard disks, like in the machine PS/1 2011 and 2121 implemented in IBMulator. 

Here is some comparison of the read/write speed of floppy drives according the 2M format software:
![2M disk speeds](https://user-images.githubusercontent.com/31600563/128944370-8072be93-c583-460e-bd92-8055ee7fa852.png)

**Comment by Mal752** [comment]
Just would like to bump this thread instead of creating a new feature request for floppy sound/noise. 

I wonder if the floppy sound emulation in Amiga emulators is accurate or not.

Please take your time. I know it's not easy and there are more important coding to do to clean up the emulator first.

The day it is implemented I will bask in nostalgia.

**Comment by ssokolow** [comment]
There's already one for floppy sound emulation. It's #460.

**Comment by ask-compu** [comment]
yes please

**Reply by parkerlreed** [reply]
Would love this

**Comment by Torinde** [comment]
Related:
- [ITX-Llama HDD activity clicker](https://www.vogons.org/viewtopic.php?p=1182888#p1182888)
> The HDD activity is controlled from hooking into disk routines in the BIOS, so I can be fancy and fine-tune the the frequency, duration, etc based on what kind of disk activity there is; reads, writes, device, lba address etc.
> going back to using the SD card's signals as the source of hdd activity for the LED and clicker. My previous attempt of letting the BIOS tell the MCU when to click worked great in DOS (which uses BIOS routines for communicating with the hard drive), but not at all in Windows

I'm not sure which approach (or a combination of those) was implemented in the later revisions.


- https://github.com/joncampbell123/dosbox-x/issues/4542
